File: Julia/sleep-sort.jl
data = [10, 4, 2, 6, 2, 7, 1, 3]
data_size = size(data)[1]
ch = Channel(data_size)

function sort_value(n)
    sleep(n / 10)
    put!(ch, n)
end

for n in data
    @async sort_value(n)
end

for _ in 1:data_size
    n = take!(ch)
    println(n)
end

File: Go/sleep-sort.go
package main

import (
	"fmt"
	"time"
)

func sort_me(ch chan int, n int) {
	time.Sleep(time.Duration(n * 100) * time.Millisecond)
	ch <- n
}

func main() {
	data := []int{10, 4, 2, 6, 2, 7, 1, 3}

	sort_channel := make(chan int, len(data))
	for _, n := range data {
		go sort_me(sort_channel, n)
	}

	for i := 0; i != len(data); i++ {
		fmt.Println(<- sort_channel)
	}
}